firmware/esp32_ttgo/client/LoRaClient.cpp

```diff
@@ -0,0 +1,40 @@
+#include "LoRaClient.h"
+
+bool LoRaClient::init()
+{
+    LL2.init();             // initialize Layer2
+    LL2.setInterval(10000); // set to zero to disable routing packets
+    return true;
+}
+
+void LoRaClient::loop()
+{
+    LL2.daemon();
+    struct Packet packet = LL2.popFromChatBuffer();
+    size_t len = packet.totalLength - HEADER_LENGTH;
+    if (packet.totalLength > HEADER_LENGTH && len > 0)
+    {
+        uint16_t msg_id;
+        char msg[len - 2 + 1] = {'\0'};
+
+        // parse out message and message id
+        memcpy(&msg_id, packet.data, 2);
+        memcpy(msg, packet.data + 2, len - 2);
+
+        server->transmit(this, String(msg));
```

Comment:
This message should not be converted to a string as it contains binary information and causes the `msg_id` issue in the WebSocketClient. Will be left in on merge, but should be fixed ASAP.

Reply:
Yeah, I'm stripping out the first 2 bytes by doing `+ 2` in the second memcpy. As we discussed, a struct containing the message ID (as `uint16_t`?), message type (as `char`?) and message payload (null terminated C string or buffer + length?) would be the right way to preserve the message ID.

Reply:
Are you saying the payload (excluding message ID) can contain binary data? In that case bytes+length rather than C string would be required.

Reply:
Yes, the payload can contain binary data. For example, the routiing table packets, which are sent from LoRaLayer2 to the WebSocket are entirely binary data